tools/CustomTasks/BatchMergeXaml.cs

```diff
@@ -0,0 +1,134 @@
+﻿using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+using System;
+using System.Collections.Generic;
+using System.IO;
+
+namespace CustomTasks
+{
+    public class BatchMergeXaml : Task
+    {
+        [Required]
+        public ITaskItem[] RS1Pages { get; set; }
+
```

Comment:
Can we just expect that the ITaskItem has attributes on it that we use to filter? I see that what you need is the API contract version, so maybe instead of the mapping in StripNamespaces we can just have the ITaskItems have the API contract version metadata directly?

Reply:
I don't put too much things in this pull request.
I was thinking about that, but the process is hard to debug, modify and publish in customtask.
Since it's very simple to prepare the parameters now in vcxproj, I still maintain split pages per release.
We can improve it in the future after compact is in

Reply:
ok

Reply:
This is for tracking purpose and I will not implement it in this PR.

possible the new signature of task in the future:
```
<BatchMergeXaml Pages="@(AllPages)" 
OverridePages="CompactPage" 
APIMinVersion=4
APIMaxVersion=9
PostfixesForGeneratedFile="generic;themeresource"
OverrideSuggestedInfix="compact"
OutputDirectory="$(OutDir)" />


```

Output file will be changed, Instead of rs2_generic.xaml, it would be api4_generic.xaml.
It would generate api?_generic.xaml and api?_compact_generic.xaml, api?_themeresources.xaml and api?_compact_ themeresources.xaml

@( AllPages) would be sorted, and then group by PostfixesForGeneratedFile="generic;themeresource", and finally apiversion.
All version tag like RS1 in Page would be changed to api version